### PR TITLE
[ur_msgs] Fix domain constants in Analog.msg

### PR DESCRIPTION
**Pull request by galou**
CURRENT and VOLTAGE should be inverted.

Tested with URSoftware 3.8.0.61336.

Signed-off-by: Gaël Écorchard gael.ecorchard@cvut.cz

### CODE REVIEW

**Comment by galou** [comment]
Cf. discussion in https://github.com/fmauch/universal_robot/pull/8.

**Comment by fmauch** [comment]
Yes, this fixes a mistake I made earlier. I checked the docs again and this seems to be the correct order.

Edit: E.g. see the [RTDE Guide](https://www.universal-robots.com/articles/ur/real-time-data-exchange-rtde-guide/) for field `analog_io_types`

**Review by fmauch — APPROVED** [review]
lgtm

**Comment by gavanderhoorn** [comment]
@galou: seems your `@cvut.cz` address is not recognised by GH.

Would you want to fix this before a merge?

**Comment by galou** [comment]
@gavanderhoorn : what do you mean "is not recognised by Github"?

**Comment by gavanderhoorn** [comment]
sshot

I would expect to see your picture there, instead we see the default GH icon.

**Comment by galou** [comment]
Actually, the guide states `{0=current[mA], 1=voltage[V]}` for the field `analog_io_types`. What I did is launching `ur_robot_driver` as of commit 936ff8d27e and looked at the topic `/joint_states` which showed a domain value of 0 when the input type was set to `Voltage` on the pendant. I think a double-check on a real robot is necessary.

**Comment by gavanderhoorn** [comment]
> looked at the topic `/joint_states`

`io_states` I guess?

**Comment by galou** [comment]
@gavanderhoorn I added the unrecognized email address to my GH settings.

**Comment by galou** [comment]
I actually meant `/ur_hardware_interface/io_states`.

**Comment by gavanderhoorn** [comment]
I won't be able to verify this with real hw for some time.

I do agree it should be verified though.

**Comment by galou** [comment]
I'm sorry, I shouldn't do two things at once...

My patch **does indeed agree** with the doc.

What I saw (last week) on the real HW is what is on the patched version, sorry for the misinformation.

**Review by gavanderhoorn — APPROVED** [review]
Approving after checking the [Real-Time Data Exchange (RTDE) Guide](https://www.universal-robots.com/articles/ur/real-time-data-exchange-rtde-guide/) document.

**Comment by gavanderhoorn** [comment]
Thanks @galou.

This will take some time to reach the repositories, as Melodic was just synced.

**Comment by gavanderhoorn** [comment]
It would be really good if we could be vocal about this.

The values in the reported IO states were correct, but the constants in the message def weren't.

However, both consumers as well as producers could be affected by this change, as they could both be using the constants and then run into issues now the values have been switched.

**Comment by fmauch** [comment]
> It would be really good if we could be vocal about this.
> 
> The values in the reported IO states were correct, but the constants in the message def weren't.
> 
> However, both consumers as well as producers could be affected by this change, as they could both be using the constants and then run into issues now the values have been switched.

Verbal where? Inside the driver, this gets read as [raw uint32_t](https://github.com/UniversalRobots/Universal_Robots_ROS_Driver/blob/master/ur_robot_driver/src/rtde/data_package.cpp#L71) and the value is [copied directly](https://github.com/UniversalRobots/Universal_Robots_ROS_Driver/blob/master/ur_robot_driver/src/ros/hardware_interface.cpp#L786) inside the hardware interface.

Do you refer to other ROS nodes as producers and consumers?

**Comment by gavanderhoorn** [comment]
> Do you refer to other ROS nodes as producers and consumers?

Yes.

**Comment by gavanderhoorn** [comment]
See [Change of constant values in ur_msgs/Analog](https://discourse.ros.org/t/change-of-constant-values-in-ur-msgs-analog/16833).

**Pull request merged**
